File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/warp/WarpService.java
package top.likoslupus.cellulosesz.api.warp;

import top.likoslupus.cellulosesz.api.teleport.CellLocation;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface WarpService {

    CompletableFuture<List<Warp>> warps();

    CompletableFuture<Optional<Warp>> warp(String name);

    CompletableFuture<Warp> setWarp(
            String name,
            CellLocation location,
            UUID creator
    );

    CompletableFuture<Boolean> deleteWarp(String name);

    CompletableFuture<Void> reload();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/warp/WarpService.kt
package top.likoslupus.cellulosesz.api.warp

import top.likoslupus.cellulosesz.api.teleport.CellLocation
import java.util.*
import java.util.concurrent.CompletableFuture

public interface WarpService {

    public fun warps(): CompletableFuture<List<Warp>>

    public fun cachedWarps(): List<Warp>

    public fun warp(name: String): CompletableFuture<Warp?>

    public fun cachedWarp(name: String): Warp?

    public fun setWarp(
        name: String,
        location: CellLocation,
        creator: UUID
    ): CompletableFuture<Warp>

    public fun deleteWarp(name: String): CompletableFuture<Boolean>

    public fun requiredPermission(warp: Warp): String?

}
